src/picongpu/include/simulationControl/MovingWindow.hpp

```diff
@@ -51,45 +51,90 @@ class MovingWindow
         if (offsetFirstGPU)
             *offsetFirstGPU = 0.0;
 
-        const SubGrid<simDim>& subGrid = Environment<simDim>::get().SubGrid();
-        const uint32_t windowGlobalDimY =
-            subGrid.getGlobalDomain().size.y() - subGrid.getLocalDomain().size.y() * slidingWindowActive;
-
-        const uint32_t devices_y = Environment<simDim>::get().GridController().getGpuNodes().y();
-        const float_64 cell_height = (float_64) CELL_HEIGHT;
-        const float_64 light_way_per_step = ((float_64) SPEED_OF_LIGHT * (float_64) DELTA_T);
-        float_64 stepsInFuture_tmp = (windowGlobalDimY * cell_height / light_way_per_step) * (1.0 - slide_point);
-        uint32_t stepsInFuture = ceil(stepsInFuture_tmp);
-        /* later used to calculate smoother offsets */
-        float_64 stepsInFutureAfterComma = stepsInFuture_tmp - (float_64) stepsInFuture;
-
-        /* round to nearest step so we get smaller sliding dfference
-         * this is valid if we activate sliding window because y direction has
-         * the same size for all gpus
-         */
-        const uint32_t stepsPerGPU = (uint32_t) math::floor(
-                                                            (float_64) (subGrid.getLocalDomain().size.y() * cell_height) / light_way_per_step + 0.5);
-        const uint32_t firstSlideStep = stepsPerGPU * devices_y - stepsInFuture;
-        const uint32_t firstMoveStep = stepsPerGPU * (devices_y - 1) - stepsInFuture;
-
-        if (slidingWindowActive==true && firstMoveStep <= currentStep)
+        if (slidingWindowActive)
         {
-            const uint32_t stepsInLastGPU = (currentStep + stepsInFuture) % stepsPerGPU;
-            /* moving window start */
-            if (firstSlideStep <= currentStep && stepsInLastGPU == 0)
-            {
-                incrementSlideCounter(currentStep);
-                if (doSlide)
-                    *doSlide = true;
-            }
+            const SubGrid<simDim>& subGrid = Environment<simDim>::get().SubGrid();
+
+            /* speed of the moving window */
+            const float_64 windowMovingSpeed = float_64(SPEED_OF_LIGHT);
+
+            /* the moving window is smaller than the global domain by exactly one
+             * GPU (local domain size)
```

Comment:
add `\todo calculation of the globalWindowSizeY is constant should be only done once in it's own central object/api`